cpp/dolfinx/mesh/GraphBuilder.cpp

```diff
@@ -271,32 +288,45 @@ compute_nonlocal_dual_graph(
             .array();
 
   // Ghost nodes: insert connected cells into local map
+  std::vector<std::vector<std::int64_t>> graph(local_graph.num_nodes());
+  for (int i = 0; i < local_graph.num_nodes(); ++i)
+  {
+    const auto& local_graph_i = local_graph.links(i);
+    for (int j = 0; j < local_graph.num_links(i); ++j)
+      graph[i].push_back(local_graph_i[j] + cell_offset);
+  }
   std::set<std::int64_t> ghost_nodes;
   std::int32_t num_nonlocal_edges = 0;
   for (int i = 0; i < cell_list.rows(); i += 2)
   {
-    assert((std::int64_t)cell_list[i] >= offset);
-    assert((std::int64_t)(cell_list[i] - offset)
-           < (std::int64_t)local_graph.size());
+    assert((std::int64_t)cell_list[i] >= cell_offset);
+    assert((std::int64_t)(cell_list[i] - cell_offset)
+           < (std::int64_t)local_graph.num_nodes());
 
-    auto& edges = graph[cell_list[i] - offset];
+    auto& edges = graph[cell_list[i] - cell_offset];
     auto it = std::find(edges.begin(), edges.end(), cell_list[i + 1]);
-    if (it == graph[cell_list[i] - offset].end())
+    if (it == graph[cell_list[i] - cell_offset].end())
     {
       edges.push_back(cell_list[i + 1]);
       ++num_nonlocal_edges;
     }
+    else
+    {
+      LOG(ERROR) << "Received same edge twice in dual graph";
+      throw std::runtime_error("Inconsistent mesh data in GraphBuilder");
```

Comment:
Make exception message more informative (like the log message).

Do we need log messages and exception messages?

Reply:
When running from python, and reading from hdf5, the code just hangs, and the runtime error never appears. 
The logging has proved very useful.